### PR TITLE
This buildpack can now use the tiny stack with reloadable processes

### PR DESCRIPTION
**Pull request by joshuatcasey**
## Summary
This buildpack can now use the tiny stack with reloadable processes. While #296 addressed this, there was some logic in the `Detect` method that was not removed.

See https://github.com/paketo-buildpacks/go-build/pull/296#issuecomment-1121189613 for additional context.

## Checklist
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by joshuatcasey** [comment]
Good catch @fg-j . "It worked on my machine!"

Will push a fix in a minute.

**Comment by joshuatcasey** [comment]
@fg-j it looks like using the tiny builder is causing further integration test problems. Should I pull it out?

I admit that this integration test would have caught the logic in `Detect` that this PR fixes, but I feel like this PR addresses that and so there's little need to make sure an integration test verifies this.

**Comment by fg-j** [comment]
Hmm... I'm concerned that that integration test isn't passing. It _should_ be, if the buildpack actually works with watchexec. This warrants more investigation.

**Comment by ForestEckhardt** [comment]
Grabbed the app locally and I get the following error 
```
[[Error (not fatal)]]
io(spawning process group): No such file or directory (os error 2)
```

**Comment by joshuatcasey** [comment]
I've completely restored the integration test to its original state. Even though this has passed on my machine it's pretty flaky even locally. 

I wonder if there's some nuance of how the builder is pulled or used that may be an issue. A quick grep through paketo codebases indicates that integration tests don't usually set the builder, although smoke tests often do (for samples and builders).

**Comment by fg-j** [comment]
@joshuatcasey You might get a clearer idea of what's failing in the container on startup if you turn up the verbosity on watchexec
See [the watchexec man page](https://github.com/watchexec/watchexec/blob/main/doc/watchexec.1.ronn#debugging-options)

**Comment by joshuatcasey** [comment]
@ForestEckhardt would you mind taking a look at the PR? I think we've gotten past the integration test issue.

File: integration/init_test.go [hunk 1]
```diff
@@ -95,6 +97,16 @@ func TestIntegration(t *testing.T) {
...
+	docker := occam.NewDocker()
+
+	t.Logf("Pulling image %s", settings.Config.TinyBuilder)
+	err = docker.Pull.Execute(settings.Config.TinyBuilder)
+	Expect(err).ToNot(HaveOccurred())
+
+	t.Logf("Pulling image %s", settings.Config.TinyRunImage)
+	err = docker.Pull.Execute(settings.Config.TinyRunImage)
+	Expect(err).ToNot(HaveOccurred())
+
```
[the remark this answers is not in the document]
Comment (fg-j) [reply integration/init_test.go:109]: Yes.

[the remark this answers is not in the document]
Comment (fg-j) [reply integration/init_test.go:109]: All of our integration tests across Paketo pull the default builder's run image for the same reason: https://github.com/paketo-buildpacks/go-build/blob/7440a55e9209b3aa6d0ee9b7d7a9da6d22bfce4d/scripts/integration.sh#L107

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**